src/wallet/wallet2.cpp

```diff
@@ -4397,6 +4673,9 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions_2(std::vector<cryp
 
     // if we need to spend money and don't have any left, we fail
     if (unused_dust_indices.empty() && unused_transfers_indices.empty()) {
+      // ToDo:
+      //    the subaddress scheme tends to fail here when the requested amount is exactly the same as the subtotal of a subaddress
+      //    because the selection of subaddr_indices doesn't take the fee into account
```

Comment:
The current code loops till it finds a way to create the tx. See wallet2.cpp:4542 for example. Are you not using the same code internally ? If so, is it broken for this particular case ? Estimating the fee is not foolproof, unfortunately, you have to build the tx then calculate to be sure. In the case above, if you're using that code with a loop, it should determine the fee is X, then determine it's not got enough, then go on adding the 2, which should be enough.